ee/models/rbac/access_control.py

```diff
@@ -0,0 +1,53 @@
+from django.db import models
+
+from posthog.models.utils import UUIDModel
+
+
+class AccessControl(UUIDModel):
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["resource", "resource_id", "team", "organization_member", "role"],
+                name="unique resource per target",
+            )
+        ]
+
+    team: models.ForeignKey = models.ForeignKey(
+        "posthog.Team",
+        on_delete=models.CASCADE,
+        related_name="access_controls",
+        related_query_name="access_controls",
+    )
+
+    # Configuration of what we are accessing
+    access_level: models.CharField = models.CharField(max_length=32)
+    resource: models.CharField = models.CharField(max_length=32)
+    resource_id: models.CharField = models.CharField(max_length=36, null=True)
+
+    # Optional scope it to a specific member
+    organization_member: models.ForeignKey = models.ForeignKey(
+        "posthog.OrganizationMembership",
+        on_delete=models.CASCADE,
+        related_name="access_controls",
+        related_query_name="access_controls",
+        null=True,
+    )
+
+    # Optional scope it to a specific role
+    role: models.ForeignKey = models.ForeignKey(
```

Comment:
these python code quality checks are weird.... Not sure if something has changed in the way we do foreign key relations? 🤔